components/org.wso2.carbon.identity.sso.samlnew/src/main/java/org/wso2/carbon/identity/sso/samlnew/bean/message/request/SAMLIdentityRequestFactory.java

```diff
@@ -0,0 +1,111 @@
+/*
+ *  Copyright (c) 2016, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ *  WSO2 Inc. licenses this file to you under the Apache License,
+ *  Version 2.0 (the "License"); you may not use this file except
+ *  in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.carbon.identity.sso.samlnew.bean.message.request;
+
+import org.apache.commons.lang.StringUtils;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.wso2.carbon.identity.application.authentication.framework.inbound.FrameworkClientException;
+import org.wso2.carbon.identity.application.authentication.framework.inbound.HttpIdentityRequestFactory;
+import org.wso2.carbon.identity.application.authentication.framework.inbound.HttpIdentityResponse;
+import org.wso2.carbon.identity.sso.samlnew.SAMLSSOConstants;
+import org.wso2.carbon.identity.sso.samlnew.exception.SAML2ClientException;
+import org.wso2.carbon.identity.sso.samlnew.util.SAMLSSOUtil;
+
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.io.UnsupportedEncodingException;
+import java.net.URLEncoder;
+import java.util.HashMap;
+import java.util.Map;
+
+public class SAMLIdentityRequestFactory extends HttpIdentityRequestFactory {
+
+    private static Log log = LogFactory.getLog(SAMLIdentityRequestFactory.class);
+    @Override
+    public String getName() {
+        return "SAMLIdentityRequestFactory";
+    }
+
+    @Override
+    public boolean canHandle(HttpServletRequest request, HttpServletResponse response) {
+        if(StringUtils.isNotBlank(request.getParameter(SAMLSSOConstants.SAML_REQUEST))){
+            return true;
+        }
+        return false;
+    }
+
+    @Override
+    public int getPriority() {
+        return -3;
+    }
+
+    @Override
+    public SAMLIdentityRequest.SAMLIdentityRequestBuilder create(HttpServletRequest request,
+                                                         HttpServletResponse response) throws SAML2ClientException {
+
+        SAMLIdentityRequest.SAMLIdentityRequestBuilder builder = new SAMLIdentityRequest.SAMLIdentityRequestBuilder
+                (request, response);
+        builder.setSamlRequest(request.getParameter(SAMLSSOConstants.SAML_REQUEST));
```

Comment:
setSamlRequest should be setSAMLRequest. Always use "SAML" instead of mixed case.